docs/ref/models/database-functions.txt

```diff
@@ -149,6 +149,41 @@ will result in a database error.
     The PostgreSQL behavior can be emulated using ``Coalesce`` if you know
     a sensible maximum value to provide as a default.
 
+``NullIf``
+----------
+
+.. class:: NullIf(expression1, expression2)
+
+.. versionadded:: 2.1
+
+Accepts two arguments, and returns ``None`` if they are equal, otherwise
+returns ``expression1``.
+
+.. admonition:: Caveats when using Oracle
+
+    Due to convention, this function returns the empty string instead of
+    ``None`` when the expressions are of type
+    :class:`~django.db.models.CharField`. See the :ref:`Oracle notes
+    <oracle-notes>` for more details.
+
+    It is not possible to pass ``Value(None)`` to ``expression1`` as Oracle
+    does not accept ``NULL`` as the first argument.
+
+Usage example::
+
+    >>> from django.db.models import Value
+    >>> from django.db.models.functions import NullIf
+    >>> Author.objects.create(name='Margaret Smith', goes_by='Maggie')
+    >>> author = Author.objects.annotate(screen_name=NullIf('goes_by', 'name')).get()
+    >>> print(author.screen_name)
+    Maggie
+
```

Comment:
Maybe add comments like some of the other examples? That being said, these examples doesn't seem particularly useful/realistic. Did you consider anything else?
```
>>> # screen_name will be None for any Authors with a goes_by equal to their name, or 'name' otherwise.
...
>>> # screen_name will be None for any Authors with a goes_by of 'Maggie', or 'goes_by' otherwise.
```

Reply:
I think this was mostly copy pasting from previous examples. `NullIf` is good for a nice trick that I read about, guarding against division by zero (`10 / nullif(0, 0)` gives `null`). And together with `Coalesce` it could also be used. I'll try to rework them.